tsconsensus: test flakes

`tsconsensus` has been flaking lately:

```
ok	tailscale.com/version/mkversion	0.004s
=== RUN   TestOnlyTaggedPeersCanDialRaftPort
=== PAUSE TestOnlyTaggedPeersCanDialRaftPort
=== CONT  TestOnlyTaggedPeersCanDialRaftPort
    tsconsensus_test.go:580: DERP httpsrv listener: 127.0.0.1:39857
    tsconsensus_test.go:580: testcontrol listening on http://127.0.0.1:43341/
2025/04/10 17:01:36 tsnet starting with hostname "node 0", varRoot "/tmp/TestOnlyTaggedPeersCanDialRaftPort3189312313/001/node 0"
2025/04/10 17:01:36 LocalBackend state is NeedsLogin; running StartLoginInteractive...
    tsconsensus_test.go:580: waitFor success: nodes tagged in status: after 2 tries
    tsconsensus_test.go:580: waitFor success: other nodes see node 1 online in ts status: after 1 tries
2025/04/10 17:01:44 Trying to find cluster: num targets to try: 2
2025/04/10 17:01:44 Trying to find cluster: trying 100.64.0.3
2025/04/10 17:01:44 Trying to find cluster: could not join 100.64.0.3: Post "http://100.64.0.3:6271/join": connect tcp 100.64.0.3:6271: connection was refused
2025/04/10 17:01:44 Trying to find cluster: trying 100.64.0.2
2025/04/10 17:01:44 Trying to find cluster: could not join 100.64.0.2: Post "http://100.64.0.2:6271/join": connect tcp 100.64.0.2:6271: connection was refused
2025/04/10 17:01:44 Trying to find cluster: unsuccessful, starting as leader: 100.64.0.1
Error: -10T17:01:44.781Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
2025-04-10T17:01:45.674Z [WARN]  raft: 2: heartbeat timeout reached, starting election: last-leader-addr=100.64.0.1:6270 last-leader-id=100.64.0.1
Error: -10T17:01:45.674Z [ERROR] raft: 2: failed to make requestVote RPC: target="{Voter 100.64.0.3 100.64.0.3:6270}" error="Get \"[http://local-tailscaled.sock/localapi/v0/status\](http://local-tailscaled.sock/localapi/v0/status/)": use of closed network connection" term=3
Error: -10T17:01:45.674Z [ERROR] raft: 2: failed to make requestVote RPC: target="{Voter 100.64.0.1 100.64.0.1:6270}" error="Get \"[http://local-tailscaled.sock/localapi/v0/status\](http://local-tailscaled.sock/localapi/v0/status/)": use of closed network connection" term=3
Error: -10T17:01:45.674Z [ERROR] raft: 2: failed to make requestVote RPC: target="{Voter 100.64.0.2 100.64.0.2:6270}" error="Get \"[http://local-tailscaled.sock/localapi/v0/status\](http://local-tailscaled.sock/localapi/v0/status/)": use of closed network connection" term=3
Error: -10T17:01:45.782Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
    tsconsensus_test.go:582: waitFor success: node 0 is leader: after 5 tries
2025/04/10 17:01:52 Trying to find cluster: num targets to try: 2
2025/04/10 17:01:52 Trying to find cluster: trying 100.64.0.1
2025/04/10 17:01:52 Trying to find cluster: joined 100.64.0.1
2025-04-10T17:01:52.735Z [WARN]  raft: 1: failed to get previous log: previous-index=3 last-index=0 error="log not found"
2025/04/10 17:01:52 Trying to find cluster: num targets to try: 2
2025/04/10 17:01:52 Trying to find cluster: trying 100.64.0.1
2025-04-10T17:01:52.736Z [WARN]  raft: 0: appendEntries rejected, sending older logs: peer="{Voter 100.64.0.2 100.64.0.2:6270}" next=1
2025-04-10T17:01:52.737Z [WARN]  raft: 2: failed to get previous log: previous-index=4 last-index=0 error="log not found"
2025/04/10 17:01:52 Trying to find cluster: joined 100.64.0.1
2025-04-10T17:01:52.737Z [WARN]  raft: 0: appendEntries rejected, sending older logs: peer="{Voter 100.64.0.3 100.64.0.3:6270}" next=1
Error: -10T17:01:52.785Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
Error: -10T17:01:53.788Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
2025-04-10T17:01:54.207Z [WARN]  raft: 2: Election timeout reached, restarting election
Error: -10T17:01:54.207Z [ERROR] raft: 2: failed to make requestVote RPC: target="{Voter 100.64.0.3 100.64.0.3:6270}" error="Get \"[http://local-tailscaled.sock/localapi/v0/status\](http://local-tailscaled.sock/localapi/v0/status/)": use of closed network connection" term=3
Error: -10T17:01:54.207Z [ERROR] raft: 2: failed to make requestVote RPC: target="{Voter 100.64.0.2 100.64.0.2:6270}" error="Get \"[http://local-tailscaled.sock/localapi/v0/status\](http://local-tailscaled.sock/localapi/v0/status/)": use of closed network connection" term=3
Error: -10T17:01:54.207Z [ERROR] raft: 2: failed to make requestVote RPC: target="{Voter 100.64.0.1 100.64.0.1:6270}" error="Get \"[http://local-tailscaled.sock/localapi/v0/status\](http://local-tailscaled.sock/localapi/v0/status/)": use of closed network connection" term=3
    tsconsensus_test.go:582: waitFor success: all raft machines have all servers in their config: after 2 tries
Error: -10T17:01:54.789Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
    tsconsensus_test.go:586: waitFor success: event makes it to all: after 2 tries
Error: -10T17:01:55.789Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
    tsconsensus_test.go:586: waitFor success: event makes it to all: after 2 tries
Error: -10T17:01:56.792Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
    tsconsensus_test.go:586: waitFor success: event makes it to all: after 2 tries
2025/04/10 17:01:57 tsnet starting with hostname "untagged node", varRoot "/tmp/TestOnlyTaggedPeersCanDialRaftPort3189312313/004/untagged node"
2025/04/10 17:01:57 LocalBackend state is NeedsLogin; running StartLoginInteractive...
Error: -10T17:01:57.793Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
Error: -10T17:01:58.794Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
Error: -10T17:01:59.794Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
Error: -10T17:02:00.796Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
2025/04/10 17:02:00 tsnet starting with hostname "untagged node", varRoot "/tmp/TestOnlyTaggedPeersCanDialRaftPort3189312313/005/untagged node"
2025/04/10 17:02:00 LocalBackend state is NeedsLogin; running StartLoginInteractive...
2025-04-10T17:02:01.089Z [WARN]  raft: 2: Election timeout reached, restarting election
Error: -10T17:02:01.089Z [ERROR] raft: 2: failed to make requestVote RPC: target="{Voter 100.64.0.3 100.64.0.3:6270}" error="Get \"[http://local-tailscaled.sock/localapi/v0/status\](http://local-tailscaled.sock/localapi/v0/status/)": use of closed network connection" term=3
Error: -10T17:02:01.089Z [ERROR] raft: 2: failed to make requestVote RPC: target="{Voter 100.64.0.1 100.64.0.1:6270}" error="Get \"[http://local-tailscaled.sock/localapi/v0/status\](http://local-tailscaled.sock/localapi/v0/status/)": use of closed network connection" term=3
Error: -10T17:02:01.089Z [ERROR] raft: 2: failed to make requestVote RPC: target="{Voter 100.64.0.2 100.64.0.2:6270}" error="Get \"[http://local-tailscaled.sock/localapi/v0/status\](http://local-tailscaled.sock/localapi/v0/status/)": use of closed network connection" term=3
Error: -10T17:02:01.796Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
2025/04/10 17:02:02 AuthLoop: state is Running; done
Error: -10T17:02:02.797Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
Error: -10T17:02:03.799Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
Error: -10T17:02:04.800Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
Error: -10T17:02:05.801Z [ERROR] raft-net: failed to accept connection: error="tsnet: use of closed network connection"
2025/04/10 17:02:05 AuthLoop: state is Running; done
    tsconsensus_test.go:592: waitFor success: nodes tagged in status: after 2 tries
2025/04/10 17:02:05 StreamLayer accept: unauthorized: 100.64.0.4:51348
    tsconsensus_test.go:623: untagged node trying to send should get a net.Error, got: EOF
2025/04/10 17:02:05 CmdHttp stopped serving with err: http: Server closed
2025/04/10 17:02:05 CmdHttp stopped serving with err: http: Server closed
2025/04/10 17:02:05 CmdHttp stopped serving with err: http: Server closed
Error: -10T17:02:05.963Z [ERROR] raft-net: failed to decode incoming command: error="read tcp 100.64.0.3:6270: operation aborted"
Error: -10T17:02:05.979Z [ERROR] raft-net: failed to decode incoming command: error="read tcp 100.64.0.2:6270: operation aborted"
    stuntest.go:65: STUN server shutdown
--- FAIL: TestOnlyTaggedPeersCanDialRaftPort (29.73s)
FAIL
FAIL	tailscale.com/tsconsensus	45.379s
```

/cc @franbull @raggi

--- a/tsconsensus/tsconsensus_test.go
+++ b/tsconsensus/tsconsensus_test.go
@@ -17,7 +17,6 @@ import (
 	"net/netip"
 	"os"
 	"path/filepath"
-	"runtime"
 	"strings"
 	"sync"
 	"testing"
@@ -27,7 +26,6 @@ import (
 	"github.com/hashicorp/go-hclog"
 	"github.com/hashicorp/raft"
 	"tailscale.com/client/tailscale"
-	"tailscale.com/cmd/testwrapper/flakytest"
 	"tailscale.com/ipn/store/mem"
 	"tailscale.com/net/netns"
 	"tailscale.com/tailcfg"
@@ -115,8 +113,8 @@ func (f *fsm) Restore(rc io.ReadCloser) error {
 }
 
 func testConfig(t *testing.T) {
-	if runtime.GOOS == "windows" && cibuild.On() {
-		t.Skip("cmd/natc isn't supported on Windows, so skipping tsconsensus tests on CI for now; see https://github.com/tailscale/tailscale/issues/16340")
+	if cibuild.On() {
+		t.Skip("these integration tests don't always work well in CI and that's bad for CI; see https://github.com/tailscale/tailscale/issues/16340 and https://github.com/tailscale/tailscale/issues/18022")
 	}
 	// -race AND Parallel makes things start to take too long.
 	if !racebuild.On {
@@ -251,7 +249,6 @@ func warnLogConfig() Config {
 }
 
 func TestStart(t *testing.T) {
-	flakytest.Mark(t, "https://github.com/tailscale/tailscale/issues/15627")
 	testConfig(t)
 	control, controlURL := startControl(t)
 	ctx := context.Background()
@@ -372,7 +369,6 @@ func createConsensusCluster(t testing.TB, ctx context.Context, clusterTag string
 }
 
 func TestApply(t *testing.T) {
-	flakytest.Mark(t, "https://github.com/tailscale/tailscale/issues/15627")
 	testConfig(t)
 	ctx := context.Background()
 	clusterTag := "tag:whatever"
@@ -437,7 +433,6 @@ func assertCommandsWorkOnAnyNode(t testing.TB, participants []*participant) {
 }
 
 func TestConfig(t *testing.T) {
-	flakytest.Mark(t, "https://github.com/tailscale/tailscale/issues/15627")
 	testConfig(t)
 	ctx := context.Background()
 	clusterTag := "tag:whatever"
@@ -477,7 +472,6 @@ func TestConfig(t *testing.T) {
 }
 
 func TestFollowerFailover(t *testing.T) {
-	flakytest.Mark(t, "https://github.com/tailscale/tailscale/issues/15627")
 	testConfig(t)
 	ctx := context.Background()
 	clusterTag := "tag:whatever"
@@ -549,7 +543,6 @@ func TestFollowerFailover(t *testing.T) {
 }
 
 func TestRejoin(t *testing.T) {
-	flakytest.Mark(t, "https://github.com/tailscale/tailscale/issues/15627")
 	testConfig(t)
 	ctx := context.Background()
 	clusterTag := "tag:whatever"
@@ -585,7 +578,6 @@ func TestRejoin(t *testing.T) {
 }
 
 func TestOnlyTaggedPeersCanDialRaftPort(t *testing.T) {
-	flakytest.Mark(t, "https://github.com/tailscale/tailscale/issues/15627")
 	testConfig(t)
 	ctx := context.Background()
 	clusterTag := "tag:whatever"
@@ -643,7 +635,6 @@ func TestOnlyTaggedPeersCanDialRaftPort(t *testing.T) {
 }
 
 func TestOnlyTaggedPeersCanBeDialed(t *testing.T) {
-	flakytest.Mark(t, "https://github.com/tailscale/tailscale/issues/15627")
 	testConfig(t)
 	ctx := context.Background()
 	clusterTag := "tag:whatever"

